### PR TITLE
Multiple fonts in lables, and width and height constraints

### LINKED ISSUE #1603: Adding font-weight to a node's label
I wanted to be able to change the font-weight of a label to be able to make it bolder. Is there already a way to manipulate this or can we add this to the package?

### LINKED ISSUE #1628: How to increase the node width
In the network graph,How would I increase the node width without increasing the white spaces in the label property.

### LINKED ISSUE #1936: How to add html text to label text?
I need to modify the label text so I can change the color for each line of the text. It is possible to achive? Adding some html or css text?

### LINKED ISSUE #2298: The size of network box could be changed without depending on text
For now, the size of the box is depends on the text inside it. What if people want to show a hiearchical tructure graph with same size of rectangle nodes? What if some of nodes only have few words inside, but the size of it should be really big to reflect some visualization parameters like number of parents?

### LINKED ISSUE #2384: Width/height options for network elements
Nodes should have minimum and maximum widths. The node's label should be able to be right- and left-padded if smaller than the minimum, and label's lines should be able to be broken on spaces if larger than the maximum.

Edges should have maximum widths. The edge's label's lines should be able to be broken on spaces if larger than the maximum.

Nodes should have minimum heights. The node's label should be top- and bottom- padded if smaller than the minimum. An alignment flag should be available to top-, middle- or bottom-justify the label when this occurs.

### PR DESCRIPTION
**Pull request by eymiha**
(This replaces PR #2362, adding functionality for width and height constraints)

fixes #1603
fixes #1628
fixes #1936
fixes #2298 
fixes #2384

Adds multifont rendering capability to network node and edge labels. Existing networks are not altered.

font.multi designates html or markdown. Bold, italic and code (monospaced) markup is supported.

font.bold designates the bold font properties, font.italic the italic font properties, font.bolditalic the bold-italic font properties, and font.mono the monospace font properties.

A vadjust property has also been added to the fonts that may be used to adjust any baseline mismatching between fonts.

The network/labels/labelMultifont.html example provides illustration and some finer details.

------

Adds minimum and maximum widths for nodes, maximum widths for edges, and minimum height and vertical label alignment for nodes. Existing networks are not altered.

Maximum width settings will break label lines on spaces to fit. Minimum width (and height) settings will expand the label horizontally (or vertically) to reach the minimum.

If the node is expanded vertically, the vertical alignment option will push it to the top, middle or bottom of the unfilled space.

The network/nodeStyles/widthHeight.html example provides illustration and some finer details.

------

Properties may be assigned in the network node and edge options, or on individual nodes and edges.

The documents are also updated.

### CODE REVIEW

**Comment by yotamberk** [comment]
@eymiha Wow! this PR is huge! Thanks for the contribution!
I've just finished going over the previous PR. Are you planning to close the first one?

**Comment by eymiha** [comment]
Yes, I just closed the first one. Sorry about the size, but I really need the functionality to move forward in some projects I was doing. It seems like the changes could benefit others as well.

**Comment by eymiha** [comment]
While eating my dog food I found a few issues and made some corrections. They were mostly edge cases, but I hit them frequently enough to require some fixing.

**Comment by yotamberk** [comment]
dog food?

**Comment by mojoaxel** [comment]
@eymiha Thanks again! Very clean and nice pull request!
I would prefer multiple smaller pull-requests next time ;-)

**Comment by eymiha** [comment]
Indeed!

File: lib/util.js [hunk 1]
```diff
@@ -1481,3 +1481,24 @@ exports.getScrollBarWidth = function () {
 
   return (w1 - w2);
 };
+
+exports.topMost = function (pile, accessors) {
```
Comment (mojoaxel) [inline lib/util.js]: @eymiha It looks like this PR broke the develop branch. Can you please checkout the current develop branch and test some simple network examples. It looks like the funktion `topMost` needs to be fixed.

File: lib/util.js [hunk 2]
```diff
@@ -1481,3 +1481,24 @@ exports.getScrollBarWidth = function () {
...
+  let candidate;
+  if (!Array.isArray(accessors)) {
+    accessors = [accessors];
+  }
+  for (const member of pile) {
+    candidate = member[accessors[0]];
```
Comment (mojoaxel) [inline lib/util.js]: @eymiha It looks like this PR broke the develop branch.

`member` can be undefined. We need a check here: 
```js
if (!member) continue;
```

Comment (mojoaxel) [inline lib/util.js]: fixed with #2403

**Pull request merged**